Tools/CISupport/Shared/maybe-delete-directory.bash

```diff
@@ -0,0 +1,45 @@
+#!/bin/bash
+
+# Deletes the directory given as the first argument if and only if it is greater than 10GB in size
+
+if [[ $# < 1 ]]
+then
+    echo "Must provide absolute path to directory to prune."
+    exit 1
+fi
+
+if [[ -L "${1%/}" ]]
+then
+    echo "$1 must not be a symbolic link."
+    exit 1
+fi
+
+if [[ -f "${1%/}" ]]
+then
+    echo "$1 must not be a file."
+    exit 1
+fi
+
+if [[ ! -d "${1%/}" ]]
+then
+    echo "$1 does not exist"
+    exit 1
+fi
+
+echo "Considering directory $1 for deletion"
+
+IFS=$'\n' read -r -d '' sizeInBytesGreaterThanThreshold < <( \
+ du -xPc $1 | \
+ awk ' { last = $0; } END { print last; } ' | \
+ awk ' { print $1; }' | \
+ awk ' { if ( $1 > 10737418240) { print "true"; } else { print "false"; } } ' && \
+ printf '\0')
```

Comment:
I don't think I'm going to be able to follow through on reviewing this, so this should be taken as a possible suggestion rather than request.

One of the things @aj062 and I have done with some more complicated buildbot steps is use `||` in a shell command to conditionally run a single command in a buildbot step without a stand-alone script. It results in the active code being quite ugly, but it's also nice because it lets us have a step that's detached from the state of the checkout and works regardless of the branch we're running on. This particular bit of code is guaranteed to have a checkout, so that isn't strictly necessary, but also, we have single check followed by a single command that is only relevant on Mac, which fits the shell-command from buildbot pattern.

Reply:
Can you point me towards some prior art demonstrating what you've described please? I'm not sure I understand the pattern you've described.

Reply:
In `Tools/CISupport/ews-build/steps.py`, if you search `self.shell_command`, you can see the places where we're running ugly compound commands. The reason I call this possibility out is that the code as proposed requires us to cherry-pick this change to a branch, probably two branches, actually. Which is certainly a thing we can do, it just increases the cost of a separate file in a buildbot step like this.

Reply:
Using [`CheckOutSpecificRevision`](https://github.com/WebKit/WebKit/blob/main/Tools/CISupport/ews-build/steps.py#L678) and the buildbot documentation as an example, I think that structure mostly makes sense. Is there a clean way to run a specific step locally? Assuming I would like to rewrite `PruneCoreSymbolicationdCacheIfTooLarge` as a `shell.Command` and run just that locally, how would I go about doing that?

With respect to unit testing, is there a preferred way to unit test `shell.Command` subclasses?

Reply:
I think you want to try and break down that script into 2 commands and join them with `&&`, the first being the check and the second being the removal. I would make sure that the combinations run locally in your terminal. Next step is running a test on our staging instance, ews-build.webkit-uat.org. That's a real buildbot instance which a few machines attached. You want to have it do a build that runs your change (cc @aj062 to help). Last step would be creating a unit test, which is mostly just going to be a string check. `CheckOutSpecificRevision` has unit tests in in `steps_unittest.py`, my suggestion is you copy and mutate those because that step has relatively simple tests.

Reply:
I've rewritten all the logic in Python instead of stretching to use shell. I tried to use `getProperty`/`setProperty` to inject fakes for `Path` and some other functionality to try to make this testable. It appears this does not work because properties have to be JSON serialisable.